File: qt/gen_qpdfoutputintent.go
package qt

import (
	"unsafe"
)

type QPdfOutputIntent struct {
	h          uintptr
	isSubclass bool
}

// NewQPdfOutputIntent constructs a new QPdfOutputIntent object.
func NewQPdfOutputIntent() *QPdfOutputIntent {

	ret := newQPdfOutputIntent(QPdfOutputIntent_new())
	ret.isSubclass = true
	return ret
}

// NewQPdfOutputIntent2 constructs a new QPdfOutputIntent object.
func NewQPdfOutputIntent2(other *QPdfOutputIntent) *QPdfOutputIntent {

	ret := newQPdfOutputIntent(QPdfOutputIntent_new2(other.cPointer()))
	ret.isSubclass = true
	return ret
}

func (this *QPdfOutputIntent) OperatorAssign(other *QPdfOutputIntent) {
	QPdfOutputIntent_OperatorAssign(this.h, other.cPointer())
}

func (this *QPdfOutputIntent) Swap(other *QPdfOutputIntent) {
	QPdfOutputIntent_Swap(this.h, other.cPointer())
}

func (this *QPdfOutputIntent) OutputConditionIdentifier() string {
	var _ms struct_miqt_string = QPdfOutputIntent_OutputConditionIdentifier(this.h)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QPdfOutputIntent) SetOutputConditionIdentifier(identifier string) {
	identifier_ms := struct_miqt_string{}
	identifier_ms.data = CString(identifier)
	identifier_ms.len = size_t(len(identifier))
	defer free(unsafe.Pointer(identifier_ms.data))
	QPdfOutputIntent_SetOutputConditionIdentifier(this.h, identifier_ms)
}

func (this *QPdfOutputIntent) OutputCondition() string {
	var _ms struct_miqt_string = QPdfOutputIntent_OutputCondition(this.h)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QPdfOutputIntent) SetOutputCondition(condition string) {
	condition_ms := struct_miqt_string{}
	condition_ms.data = CString(condition)
	condition_ms.len = size_t(len(condition))
	defer free(unsafe.Pointer(condition_ms.data))
	QPdfOutputIntent_SetOutputCondition(this.h, condition_ms)
}

func (this *QPdfOutputIntent) RegistryName() *QUrl {
	_goptr := newQUrl(QPdfOutputIntent_RegistryName(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QPdfOutputIntent) SetRegistryName(name *QUrl) {
	QPdfOutputIntent_SetRegistryName(this.h, name.cPointer())
}

func (this *QPdfOutputIntent) OutputProfile() *QColorSpace {
	_goptr := newQColorSpace(QPdfOutputIntent_OutputProfile(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QPdfOutputIntent) SetOutputProfile(profile *QColorSpace) {
	QPdfOutputIntent_SetOutputProfile(this.h, profile.cPointer())
}

File: qt/gen_qpdfoutputintent.cpp
// +build ignore

#include <QColorSpace>
#include <QPdfOutputIntent>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QUrl>
#include <qpdfoutputintent.h>
#include "gen_qpdfoutputintent.h"

#ifndef _Bool
#define _Bool bool
#endif

QPdfOutputIntent* QPdfOutputIntent_new() {
	return new QPdfOutputIntent();
}

QPdfOutputIntent* QPdfOutputIntent_new2(QPdfOutputIntent* other) {
	return new QPdfOutputIntent(*other);
}

void QPdfOutputIntent_OperatorAssign(QPdfOutputIntent* self, QPdfOutputIntent* other) {
	self->operator=(*other);
}

void QPdfOutputIntent_Swap(QPdfOutputIntent* self, QPdfOutputIntent* other) {
	self->swap(*other);
}

struct miqt_string QPdfOutputIntent_OutputConditionIdentifier(const QPdfOutputIntent* self) {
	QString _ret = self->outputConditionIdentifier();
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

void QPdfOutputIntent_SetOutputConditionIdentifier(QPdfOutputIntent* self, struct miqt_string identifier) {
	QString identifier_QString = QString::fromUtf8(identifier.data, identifier.len);
	self->setOutputConditionIdentifier(identifier_QString);
}

struct miqt_string QPdfOutputIntent_OutputCondition(const QPdfOutputIntent* self) {
	QString _ret = self->outputCondition();
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

void QPdfOutputIntent_SetOutputCondition(QPdfOutputIntent* self, struct miqt_string condition) {
	QString condition_QString = QString::fromUtf8(condition.data, condition.len);
	self->setOutputCondition(condition_QString);
}

QUrl* QPdfOutputIntent_RegistryName(const QPdfOutputIntent* self) {
	return new QUrl(self->registryName());
}

void QPdfOutputIntent_SetRegistryName(QPdfOutputIntent* self, QUrl* name) {
	self->setRegistryName(*name);
}

QColorSpace* QPdfOutputIntent_OutputProfile(const QPdfOutputIntent* self) {
	return new QColorSpace(self->outputProfile());
}

void QPdfOutputIntent_SetOutputProfile(QPdfOutputIntent* self, QColorSpace* profile) {
	self->setOutputProfile(*profile);
}

void QPdfOutputIntent_Delete(QPdfOutputIntent* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QPdfOutputIntent*>( self );
	} else {
		delete self;
	}
}

